File: testdata/PullUpField/Java/parent/Dog.java
public class Dog {
    protected String name;

    public Dog(String name) {
        this.name = name;
    }

    public void bark() {
        System.out.println(name + " is barking!");
    }
}

public class Poodle extends Dog {

    public Poodle(String name) {
        super(name);
    }
}

File: testdata/PullUpField/Cpp/parent/Dog.cpp
#include <iostream>
#include <string>

class Dog {
protected:
    std::string name;

public:
    Dog(const std::string& name) : name(name) {}

    void bark() {
        std::cout << name << " is barking!" << std::endl;
    }
};

class Poodle : public Dog {
public:
    Poodle(const std::string& name) : Dog(name) {}
};
